public/js/extended-access.js

```diff
@@ -0,0 +1,240 @@
+      resolve(userState);
+    });
+  });
+
+  // Get userState by passing the gaaUser object to your registration endpoint server
+  // Resolves with the current user’s entitlements from the publisher
+  // Optional - required only if your userState object doesn't contain the publisherEntitlement already
+  // Do not copy the implementation for the promise and see documentation for your own implementation at:
+  // https://developers.google.com/news/subscribe/extended-access/integration-steps/web-implementation#creating-the-users-entitlement-state
+  const publisherEntitlementPromise = new Promise ((resolve) => {
+    // Example body of the promise returning a publisherEntitlement object
+    const publisherEntitlement = {
+      granted: false,
+    };
+    resolve(publisherEntitlement);
+  });
+
+  // The userState object for the current user.
+  // Do not copy the implementation for the callback and see documentation for your own implementation at:
+  // https://developers.google.com/news/subscribe/extended-access/reference/user-state-object
+  function getUserState() {
+    return {
+      // Commented out since we don't actually use the UserState in this demo of the
+      // registration flow.
+      // id: 'user123456789',
+      // registrationTimestamp: 1647311022,
+      // subscriptionTimestamp: 1647311062,
+      // granted: true,
+      // grantReason: 'SUBSCRIBER'
+    };
+  };
+
+  /* Code */
+
+  // Provide a shortcut to metering params.
+  // Do not copy the constant and see documentation at:
+  // https://developers.google.com/news/subscribe/extended-access/reference/google-article-access-parameters
+  const urlParams = new URLSearchParams(location.search);
+  if (urlParams.has('metering')) {
+    let newSearch = 'gaa_at=g&gaa_ts=99999999&gaa_n=n0nc3&gaa_sig=51g';
+    location.search = newSearch;
+  }
+
+
+  // Array with all allowed hosts that can refer users to your article pages.
+  // This is used if you redirect users or refresh the page during the Handle registration
+  // for new users or Handle login for existing users flows.
+  // Do not copy the constant and see documentation at:
+  // https://developers.google.com/news/subscribe/extended-access/integration-steps/web-implementation#initialize-the-extended-access-library
+  const allowedReferrers = ["ea-upgrade-dot-reader-revenue-demo.ue.r.appspot.com",
+                            "ea-upgrade-to-rrme-dot-reader-revenue-demo.ue.r.appspot.com",
+                            "b2607f8b04800100000144d90c0a821641f90000000000000000001.proxy.googlers.com"]
+
+  // User's Showcase entitlement determined on your server
+  // Optional - server side only
+  // Do not copy the constant and see documentation at:
+  // https://developers.google.com/news/subscribe/guides/check-for-entitlements-on-web
+  const showcaseEntitlement = urlParams.get("showcaseEntitlement");
```

Comment:
The rest of the demo is client-side. Maybe remove this and create a separate server-side demo?

Reply:
Removed.